### PR TITLE
Sync ts logic to c++

### PR DESCRIPTION
**Pull request by minggo**

### CODE REVIEW

**Comment by minggo** [comment]
@jk20012001 please help to review skybox related codes
@yiwenxue please help to review mipmap related codes
@dumganhar please help to review model related codes

**Review by dumganhar — APPROVED** [review]

File: native/cocos/core/assets/SimpleTexture.cpp [hunk 1]
```diff
@@ -159,23 +159,24 @@ void SimpleTexture::createTexture(gfx::Device *device) {
...
+    maxLevel - _baseLevel + 1
     );
 
     //TODO(minggo)
 //    if (!textureViewCreateInfo) {
 //        return;
 //    }
     
-    _gfxTextureView = device->createTexture(textureViewCreateInfo);
+    return device->createTexture(textureViewCreateInfo);
```
Comment (yiwenxue) [inline native/cocos/core/assets/SimpleTexture.cpp:179]: We should also free their memory when destroying `_gfxTexture` and `_gfxTextureView`.

Comment (minggo) [reply native/cocos/core/assets/SimpleTexture.cpp:179]: It is destroyed in `SimpleTexture::tryDestroyTextureView`. And it is the same logic as TS.

**Review by jk20012001 — APPROVED** [review]

File: native/cocos/core/assets/SimpleTexture.cpp (hunk 1 above)
Comment (yiwenxue) [reply native/cocos/core/assets/SimpleTexture.cpp:179]: Yes, exactly, Typescript has a GC system, and the memory is freed automatically. But for C++, things are different, `_gfxTexture->destroy()` will only free the resource on GPU, and we need to call `CC_DELETE(_gfxTexture)` mannually.

Comment (minggo) [reply native/cocos/core/assets/SimpleTexture.cpp:179]: They are IntrusivePtr, destructor will be invoked automatically.

Comment (yiwenxue) [reply native/cocos/core/assets/SimpleTexture.cpp:179]: Ok, fine.

**Review by yiwenxue — APPROVED** [review]

**Pull request merged**